### ISSUE TITLE
Optimizer Params in GCN Example

### ISSUE THREAD

**Issue by ilteralp** [pyg-team/pytorch_geometric#1468]
## ❓ Questions & Help

Hi all, I'm trying to understand [gcn.py](https://github.com/rusty1s/pytorch_geometric/blob/eca4f44052081760f365da5f9fc679743c0558a3/examples/gcn.py) example. I have a question. Why is `model.non_reg_params` used in optimizer ? Also, what does it stand for ? I checked Kipf's paper and could find it there. I also checked examples and could not find a similar approach. Thank you.

**Comment by rusty1s** [comment]
That means that we only want to apply weight decay on the first convolution, and not the one that performs the actual prediction. This is in-line with the official implementation, see [here](https://github.com/tkipf/gcn/blob/dce3737f5e4363b8bd1da0a450d4a6ea58455bc3/gcn/models.py#L101).
